2018/proposals/gensim Aneesh Joshi Similarity Learning.md

```diff
@@ -0,0 +1,374 @@
+  	 <td align="center", bgcolor=#eeeeee> 0.5988 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> DRMM </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6107 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6621 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6195 </td>
+  </tr>
+  <tr>
+     <td align="center", bgcolor=#eeeeee> aNMM </td>
+     <td align="center", bgcolor=#eeeeee> 0.6160 </td>
+     <td align="center", bgcolor=#eeeeee> 0.6696 </td>
+     <td align="center", bgcolor=#eeeeee> 0.6297 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> DUET </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6065 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6722 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6301 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> MatchPyramid </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6317 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6913 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6434 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> DRMM_TKS </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6458 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6956 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6586 </td>
+  </tr>
+  <tr>
+  	 <td align="center", bgcolor=#eeeeee> K-NRM </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6268 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6693 </td>
+  	 <td align="center", bgcolor=#eeeeee> 0.6256 </td>
+  </tr>
+</table>
+
+Along with the loss:
+
+![alt-text](https://github.com/faneshion/MatchZoo/blob/master/docs/_static/images/matchzoo.wikiqa.loss.png)
+
+Free datasets for Similarity Learning will also be added to [gensim-data](https://github.com/RaRe-Technologies/gensim-data) :
+
+* [Quora Duplicate Questions](https://data.quora.com/First-Quora-Dataset-Release-Question-Pairs) (already in gensim-data)
+* [WikiQA](https://www.microsoft.com/en-us/download/details.aspx?id=52419)
+* [Ubuntu Dialogue Corpus](https://github.com/rkadlec/ubuntu-ranking-dataset-creator)
+
+# Deliverables
+
+The whole project can be broken down into the following phases:
+1. Setup of evaluation scheme for all the models which will be introduced
+2. Collect/Implement models for comparisons, benchmarking and parameter tuning
+3. Choose the best models and integrate them into gensim
+4. Work on improving model
+
+### Community Bonding Period (April 23 to May 14)
+* Read papers relating to Similarity Learning
+* Work on ongoing PRs
+* Address existing issues and doubts on gitter and stack overflow
+
+
+### Week 1 (May 14 to May 20)
+* Discuss and create an evaluation pipeline for possible models to be included
+* Evaluate current list of models to be implemented and make possible additions/deletions
+
+(It's possible that the list of models may change at this point and accomodations will be made for the same.)
+
+### Week 2 (May 21 to May 27)
+* Develop or use existing implementation of DSSM, CDSSM, MV-LSTM

```

Comment:
Strange order: you evaluate models on previous, but now - "develop or use existing implementations", how it possible?

Reply:
What I meant was:
In the previous paragraph (i.e., in the community bonding period), I will set up an evaluation pipeline which will be applied to all the models which are developed.

In the end, we will have the outcome of the evaluation on each of the models and then pick the best ones. (around week 5)